### ISSUE TITLE
Discount application ordering and priority

### ISSUE THREAD

**Issue by chrisvidal** [OFFLINE-GmbH/oc-mall-plugin#888]
I think this is a problem with ordering and priority of application of discounts.

Let's say we have a discount 'FREE SHIPPING' for any order total above 50$.
Let's say we have a cart containing items for a total price of 55$.
So the free shipping will apply.
Let's say the customer apply a discount code giving him back 10$.
So the new order total is 45$.

But the FREE SHIPPING stills applies automatically, and I guess it shouldn't since the total is now 45$.

EDIT: using `MALL 1.13.5` and October v1 (472)

**Comment by martinkuch** [comment]
Discount coupons are calculated for products total price not for whole order, I believe.

**Comment by chrisvidal** [comment]
yes indeed but this is not the issue at play here. The issue is that the alternate shipping still applies when it should not.

**Comment by chrisvidal** [comment]
my bad, I just realized my mistake.

It is not an alternate shipping discount that is applied, it is a shipping method that becomes available when the order is above a certain amount.

So, still the problem appears as the shipping method appears even after the application of the discount code.

![image](https://user-images.githubusercontent.com/2263496/145147044-2361e72e-ac1a-451f-8dc1-7aab8bee7e62.png)

**Comment by chrisvidal** [comment]
any updates on this please?

**Comment by georgij-** [comment]
Hi @chrisvidal 

I've had a similar issue, ended up modifying this line of code here,

https://github.com/OFFLINE-GmbH/oc-mall-plugin/blob/98a5028b854699fca1bf7e07e36be9a6cd13db8c/models/ShippingMethod.php#L175

Where I've subtracted the discount total, so the correct shipping method will return.

**Comment by tobias-kuendig** [comment]
This is fixed in v3.6.1
